packages/tupaia-web/src/features/Map/utils/useMapOverlayData.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import { MeasureData, Series } from '@tupaia/ui-map-components';
+import { LegendProps } from '@tupaia/ui-map-components';
+import { useEntitiesWithLocation } from '../../../api/queries';
+import { processMeasureData } from '../MapOverlays/processMeasureData';
+import { useMapOverlays } from '../../../api/queries';
+import { useMapOverlayReport } from '../../../api/queries';
+import { EntityCode, ProjectCode } from '../../../types';
+
+export const useMapOverlayData = (
+  projectCode?: ProjectCode,
+  entityCode?: EntityCode,
+  hiddenValues?: LegendProps['hiddenValues'],
+): { serieses: Series[] | undefined; processedMeasureData: MeasureData[] | undefined } => {
+  const useEntitiesByMeasureLevel = (measureLevel?: string) => {
+    const getSnakeCase = (measureLevel?: string) => {
```

Comment:
Just for the sake of keeping this tidy, I think `getSnakeCase` can move outside of the hook. It can stay in this file but just as a standalone function